src/darwin/Framework/CHIP/MTRDeviceControllerStorageDelegate.h

```diff
@@ -111,6 +111,34 @@ MTR_NEWLY_AVAILABLE
     removeValueForKey:(NSString *)key
         securityLevel:(MTRStorageSecurityLevel)securityLevel
           sharingType:(MTRStorageSharingType)sharingType;
+
+@optional
+/**
+ * Return all keys and values stored, if any, for the provided controller in a
+ * dictionary. Returns nil if there is no stored value.
+ *
+ * securityLevel and dataType will always be the same for any given key value
+ * and are just present here to help locate the data if storage location is
+ * separated out by security level and data type.
```

Comment:
I don't see how this works.  What happens if in fact data with different security levels and sharing types (not dataType? I guess we have that mis-naming in all the other comments too, and should fix it) is stored for the controller?

Reply:
We are of course also stating that for any given key, the consumer should not expect the securityLevel and sharingType to change. But I'm not sure there is much to enforce/check here. The framework is making a promise and should not break it.

If we break the promise, and the API consumer in fact does have separate storage for each securityLevel/sharingType, then it'll be a framework bug that needs to be fixed.

Reply:
My point is this: we are asking for all the keys.  But we are also providing the dataType and sharingType.

Does this actually mean that we are asking just for all the keys _for that dataType and sharingType_?  If so, we should clearly say that.

Reply:
I've consolidated the description and clarified a bit.